### PR TITLE
[proj4]Upgrade version to 6.1.1 and remove useless patches.

### PR DESCRIPTION
**Pull request by JackBoosY**
1. Upgrade proj4 to v6.1.1 
2. Change the source to github.
3. Add a new version of the dependency port `sqlite3[tool]`.
4. Remove useless patches.
5. According to the official [response](https://github.com/OSGeo/PROJ/issues/1479#issuecomment-515925289), change the generated library name to `proj.lib`.
6. Feature `database` need to install sqlite3:x86-windows/sqlite3:x64-windows on triplet arm/arm64/uwp.
7. Fix the default library name of proj4 used by gdal.

Related: #3258 #3408 #6180 #6949..

### CODE REVIEW

File: ports/liblas/portfile.cmake [hunk 1]
```diff
@@ -15,6 +15,18 @@ vcpkg_extract_source_archive_ex(
     PATCHES fix-BuildError.patch
 )
 
+if (VCPKG_TARGET_IS_WINDOWS)
```
Comment (Neumann-A) [inline ports/liblas/portfile.cmake]: find_library actually works now in portfiles. so you could use something like:

```
find_library(SQLITE_RELEASE NAMES sqlite3 PATHS "${CURRENT_INSTALLED_DIR}/lib" NO_DEFAULT_PATH) # Depends on openssl and zlib(linux)
find_library(SQLITE_DEBUG NAMES sqlite3 sqlite3d PATHS "${CURRENT_INSTALLED_DIR}/debug/lib" NO_DEFAULT_PATH)
```

File: ports/gdal/portfile.cmake [hunk 2]
```diff
@@ -252,6 +252,7 @@ if (NOT VCPKG_CMAKE_SYSTEM_NAME OR VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStor
     # Release build
     ################
     message(STATUS "Building ${TARGET_TRIPLET}-rel")
+    message(STATUS "Release Options: ${NMAKE_OPTIONS_REL}")
```
Comment (ras0219-msft) [inline ports/gdal/portfile.cmake]: Leftover debug messages

File: ports/liblas/fix-BuildError.patch [hunk 3]
```diff
@@ -41,7 +41,7 @@ index e35de06..f7c25d2 100644
    ${LASZIP_LIBRARY}
 -  ${Boost_LIBRARIES})
 +  ${Boost_LIBRARIES}
-+  proj)
++  ${PROJ4_LIBRARIES} ${SQLITE3_LIBRARIES})
```
Comment (ras0219-msft) [inline ports/liblas/fix-BuildError.patch]: Why does this suddenly need sqlite3?

If this is a transitive dependency due to proj4, we should push this into the proj4 config files.

File: ports/sqlite3/CMakeLists.txt [hunk 4]
```diff
@@ -27,7 +27,7 @@ target_compile_definitions(
 target_include_directories(sqlite3 INTERFACE $<INSTALL_INTERFACE:include>)
 if(NOT WIN32)
     find_package(Threads REQUIRED)
-    target_link_libraries(sqlite3 PRIVATE Threads::Threads ${CMAKE_DL_LIBS})
+    target_link_libraries(sqlite3 PRIVATE Threads::Threads PUBLIC ${CMAKE_DL_LIBS})
```
Comment (ras0219-msft) [inline ports/sqlite3/CMakeLists.txt]: Sqlite doesn't use DL_LIBS in its public interface -- someone else in the chain must be causing any problems that this fixes.

**Comment by JackBoosY** [comment]
@ras0219-msft Thanks for review.

**Comment by JackBoosY** [comment]
`otl`, `hungarian` and `libxmp-lite` fails to download its source.
Seems CI machine cannot connect to source website.

LLVM has copy file error,  I cannot repro it on my machine.

**Comment by JackBoosY** [comment]
Only have OSX failure here:
```
FAILED: bin/Debug/liblas.2.4.0.dylib 
: && /Library/Developer/CommandLineTools/usr/bin/c++ -fPIC  -Wextra -Wall -Wno-unused-parameter -Wno-unused-variable -Wpointer-arith -Wcast-align -Wcast-qual -Wfloat-equal -Wredundant-decls -Wno-long-long -g  -dynamiclib -Wl,-headerpad_max_install_names -compatibility_version 3.0.0 -current_version 2.4.0 -o bin/Debug/liblas.2.4.0.dylib -install_name @rpath/liblas.3.dylib src/CMakeFiles/las.dir/detail/index/indexcell.cpp.o src/CMakeFiles/las.dir/detail/index/indexoutput.cpp.o src/CMakeFiles/las.dir/chipper.cpp.o src/CMakeFiles/las.dir/factory.cpp.o src/CMakeFiles/las.dir/classification.cpp.o src/CMakeFiles/las.dir/color.cpp.o src/CMakeFiles/las.dir/dimension.cpp.o src/CMakeFiles/las.dir/error.cpp.o src/CMakeFiles/las.dir/filter.cpp.o src/CMakeFiles/las.dir/header.cpp.o src/CMakeFiles/las.dir/index.cpp.o src/CMakeFiles/las.dir/point.cpp.o src/CMakeFiles/las.dir/reader.cpp.o src/CMakeFiles/las.dir/spatialreference.cpp.o src/CMakeFiles/las.dir/schema.cpp.o src/CMakeFiles/las.dir/transform.cpp.o src/CMakeFiles/las.dir/utility.cpp.o src/CMakeFiles/las.dir/variablerecord.cpp.o src/CMakeFiles/las.dir/writer.cpp.o src/CMakeFiles/las.dir/version.cpp.o src/CMakeFiles/las.dir/detail/utility.cpp.o src/CMakeFiles/las.dir/detail/sha1.cpp.o src/CMakeFiles/las.dir/detail/zippoint.cpp.o src/CMakeFiles/las.dir/detail/reader/header.cpp.o src/CMakeFiles/las.dir/detail/reader/reader.cpp.o src/CMakeFiles/las.dir/detail/reader/zipreader.cpp.o src/CMakeFiles/las.dir/detail/reader/cachedreader.cpp.o src/CMakeFiles/las.dir/detail/writer/header.cpp.o src/CMakeFiles/las.dir/detail/writer/point.cpp.o src/CMakeFiles/las.dir/detail/writer/zipwriter.cpp.o src/CMakeFiles/las.dir/detail/writer/writer.cpp.o -L/Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libtiffd.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/liblzmad.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libjpeg.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libz.a -lm /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libgeotiff.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libboost_program_options.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libboost_thread.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libboost_system.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libboost_iostreams.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libboost_filesystem.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libboost_chrono.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libboost_date_time.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libboost_atomic.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libboost_regex.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libtiffd.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/liblzmad.a -lproj /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libjpeg.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libz.a -lm && :
Undefined symbols for architecture x86_64:
  "_sqlite3_bind_double", referenced from:
      osgeo::proj::io::DatabaseContext::Private::run(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::list<osgeo::proj::io::SQLValues, std::__1::allocator<osgeo::proj::io::SQLValues> > const&) in libproj.a(factory.cpp.o)
  "_sqlite3_bind_text", referenced from:
      osgeo::proj::io::DatabaseContext::Private::run(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::list<osgeo::proj::io::SQLValues, std::__1::allocator<osgeo::proj::io::SQLValues> > const&) in libproj.a(factory.cpp.o)
  "_sqlite3_close", referenced from:
      osgeo::proj::io::DatabaseContext::Private::closeDB() in libproj.a(factory.cpp.o)
  "_sqlite3_column_count", referenced from:
      osgeo::proj::io::DatabaseContext::Private::run(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::list<osgeo::proj::io::SQLValues, std::__1::allocator<osgeo::proj::io::SQLValues> > const&) in libproj.a(factory.cpp.o)
  "_sqlite3_column_text", referenced from:
      osgeo::proj::io::DatabaseContext::Private::run(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::list<osgeo::proj::io::SQLValues, std::__1::allocator<osgeo::proj::io::SQLValues> > const&) in libproj.a(factory.cpp.o)
  "_sqlite3_create_function", referenced from:
      osgeo::proj::io::DatabaseContext::Private::registerFunctions() in libproj.a(factory.cpp.o)
  "_sqlite3_errmsg", referenced from:
      osgeo::proj::io::DatabaseContext::Private::run(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::list<osgeo::proj::io::SQLValues, std::__1::allocator<osgeo::proj::io::SQLValues> > const&) in libproj.a(factory.cpp.o)
  "_sqlite3_finalize", referenced from:
      osgeo::proj::io::DatabaseContext::Private::closeDB() in libproj.a(factory.cpp.o)
  "_sqlite3_open_v2", referenced from:
      osgeo::proj::io::DatabaseContext::Private::open(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, projCtx_t*) in libproj.a(factory.cpp.o)
      osgeo::proj::io::DatabaseContext::Private::attachExtraDatabases(std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) in libproj.a(factory.cpp.o)
  "_sqlite3_prepare_v2", referenced from:
      osgeo::proj::io::DatabaseContext::Private::run(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::list<osgeo::proj::io::SQLValues, std::__1::allocator<osgeo::proj::io::SQLValues> > const&) in libproj.a(factory.cpp.o)
  "_sqlite3_reset", referenced from:
      osgeo::proj::io::DatabaseContext::Private::run(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::list<osgeo::proj::io::SQLValues, std::__1::allocator<osgeo::proj::io::SQLValues> > const&) in libproj.a(factory.cpp.o)
  "_sqlite3_result_double", referenced from:
      osgeo::proj::io::PROJ_SQLITE_pseudo_area_from_swne(sqlite3_context*, int, sqlite3_value**) in libproj.a(factory.cpp.o)
  "_sqlite3_result_int", referenced from:
      osgeo::proj::io::PROJ_SQLITE_intersects_bbox(sqlite3_context*, int, sqlite3_value**) in libproj.a(factory.cpp.o)
  "_sqlite3_result_null", referenced from:
      osgeo::proj::io::PROJ_SQLITE_pseudo_area_from_swne(sqlite3_context*, int, sqlite3_value**) in libproj.a(factory.cpp.o)
      osgeo::proj::io::PROJ_SQLITE_intersects_bbox(sqlite3_context*, int, sqlite3_value**) in libproj.a(factory.cpp.o)
  "_sqlite3_snprintf", referenced from:
      osgeo::proj::internal::toString(int) in libproj.a(internal.cpp.o)
      osgeo::proj::internal::toString(double, int) in libproj.a(internal.cpp.o)
  "_sqlite3_step", referenced from:
      osgeo::proj::io::DatabaseContext::Private::run(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::list<osgeo::proj::io::SQLValues, std::__1::allocator<osgeo::proj::io::SQLValues> > const&) in libproj.a(factory.cpp.o)
  "_sqlite3_value_double", referenced from:
      osgeo::proj::io::PROJ_SQLITE_GetValAsDouble(sqlite3_value*, bool&) in libproj.a(factory.cpp.o)
  "_sqlite3_value_int64", referenced from:
      osgeo::proj::io::PROJ_SQLITE_GetValAsDouble(sqlite3_value*, bool&) in libproj.a(factory.cpp.o)
  "_sqlite3_value_type", referenced from:
      osgeo::proj::io::PROJ_SQLITE_GetValAsDouble(sqlite3_value*, bool&) in libproj.a(factory.cpp.o)
  "_sqlite3_vfs_find", referenced from:
      osgeo::proj::io::DatabaseContext::Private::createCustomVFS() in libproj.a(factory.cpp.o)
      osgeo::proj::io::VFSClose(sqlite3_file*) in libproj.a(factory.cpp.o)
  "_sqlite3_vfs_register", referenced from:
      osgeo::proj::io::DatabaseContext::Private::createCustomVFS() in libproj.a(factory.cpp.o)
  "_sqlite3_vfs_unregister", referenced from:
      osgeo::proj::io::DatabaseContext::Private::~Private() in libproj.a(factory.cpp.o)
ld: symbol(s) not found for architecture x86_64
```

I don't have osx machine, can anyone help me?

**Comment by JackBoosY** [comment]
@dan-shaw  I can't reproduce the qt5 component build failure on my machine. Can you test them?
If it is fake, this PR should be merged.

**Comment by dan-shaw** [comment]
Besides the `ports/sqlite3/CONTROL` merge conflict, LGTM.

**Comment by JackBoosY** [comment]
Already merged 9 times...OMG..

**Pull request merged**